# 🏥 Healthcare Patient Analytics & Risk Analysis System

## Phase 2 — Day 3: Data Cleaning

### Objective

The objective of Day 3 is to identify and handle missing
values in the healthcare patient dataset.

We will:

- Identify missing values
- Calculate missing percentages
- Understand why missing values matter
- Handle numerical missing values
- Handle categorical missing values
- Verify the cleaned dataset

In [1]:
#  import libraries
import pandas as pd
import numpy as np

print("Libraries imported successfully.")

Libraries imported successfully.


In [2]:
# load dataset
df = pd.read_csv("/content/healthcare_patients.csv")

print("Dataset loaded successfully.")

Dataset loaded successfully.


In [3]:
df.shape

(10020, 16)

In [11]:
# check missing values
df.isnull().sum()

,0
patient_id,0
age,0
gender,0
blood_pressure,150
heart_rate,120
bmi,180
smoking_status,100
diabetes,0
hypertension,0
disease,0


In [13]:
# calculate missing percentage
missing_percentage = ((df.isnull().sum() / len(df)) *100 ).round(2)
missing_percentage[missing_percentage>0]

,0
blood_pressure,1.5
heart_rate,1.2
bmi,1.8
smoking_status,1.0
medication_count,1.2
treatment_cost,0.8


In [17]:
# Create a missing-value summary
missing_summary = pd.DataFrame({
    'Missing Values': df.isnull().sum(),
    'Missing Percentage': missing_percentage
})
missing_summary = missing_summary[missing_summary['Missing Values'] > 0]
missing_summary

,Missing Values,Missing Percentage
blood_pressure,150,1.5
heart_rate,120,1.2
bmi,180,1.8
smoking_status,100,1.0
medication_count,120,1.2
treatment_cost,80,0.8


In [18]:
# finding the meadian of numearical missing values
numerical_missing_columns = [
    "blood_pressure",
    "heart_rate",
    "bmi",
    "medication_count",
    "treatment_cost"
]

df[numerical_missing_columns].median()

,0
blood_pressure,128.0
heart_rate,78.0
bmi,26.3
medication_count,3.0
treatment_cost,27953.0


In [20]:
# fill missing values with meadian
for column in numerical_missing_columns:
    df[column] = df[column].fillna(df[column].median())

df[numerical_missing_columns].isnull().sum()

,0
blood_pressure,0
heart_rate,0
bmi,0
medication_count,0
treatment_cost,0


In [21]:
# Handle categorical missing value
df["smoking_status"].value_counts()

,count
smoking_status,
Never,5716
Former,2466
Current,1738


In [22]:
df["smoking_status"].value_counts(dropna=False)

,count
smoking_status,
Never,5716
Former,2466
Current,1738
NaN,100


In [28]:
# Fill categorical missing values
df["smoking_status"].mode()[0]
df["smoking_status"] = df["smoking_status"].fillna(
    df["smoking_status"].mode()[0]
)

In [30]:
# verify all missing values
df.isnull().sum()
df.isnull().sum().sum()

np.int64(0)

In [31]:
# verify data set
df.shape

(10020, 16)

In [32]:
# Create a cleaning summary
cleaning_summary = pd.DataFrame({
    "column": [
        "blood_pressure",
        "heart_rate",
        "bmi",
        "medication_count",
        "treatment_cost",
        "smoking_status"
    ],
    "data_type": [
        "Numerical",
        "Numerical",
        "Numerical",
        "Numerical",
        "Numerical",
        "Categorical"
    ],
    "missing_value_strategy": [
        "Median",
        "Median",
        "Median",
        "Median",
        "Median",
        "Mode"
    ]
})

cleaning_summary

,column,data_type,missing_value_strategy
0,blood_pressure,Numerical,Median
1,heart_rate,Numerical,Median
2,bmi,Numerical,Median
3,medication_count,Numerical,Median
4,treatment_cost,Numerical,Median
5,smoking_status,Categorical,Mode


# 📋 Day 3 Observations

## Missing Values

The original dataset contained 750 missing cells across
six columns.

Missing values were found in:

- blood_pressure
- heart_rate
- bmi
- smoking_status
- medication_count
- treatment_cost

## Missing Value Strategy

### Numerical Variables

Missing values in:

- blood_pressure
- heart_rate
- bmi
- medication_count
- treatment_cost

were replaced using the respective column median.

### Categorical Variables

Missing values in:

- smoking_status

were replaced using the mode of the column.

## Verification

After imputation:

- Total missing values: 0
- Dataset rows: 10,020
- Dataset columns: 16

No patient records were removed during missing-value handling.

## Conclusion

The missing-value problem has been addressed while preserving
the original number of patient records.

The cleaned DataFrame is now ready for the next stage of
data cleaning.